internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -1339,13 +1341,23 @@ func (c *container) addUserbindsMount(system *mount.System) error {
 
 		sylog.Debugf("Adding %s to mount list\n", src)
 
+		if c.userNS {
+			// test if the source directory is a read-only filesystem by
+			// asking for write permission for this directory, if EROFS error
+			// is returned, we need to skip remount to avoid failure
+			if err := syscall.Access(src, 0x2); err.(syscall.Errno) == syscall.EROFS {
+				remount = false
+			}
+		}
+
 		if err := system.Points.AddBind(mount.UserbindsTag, src, dst, flags); err != nil && err == mount.ErrMountExists {
```

Comment:
This comment is for the unchanged AddBind() line: the err != nil check is redundant, because if it is equal to mount.ErrMountExists it is obviously not nil.